docs/configuration/index.md

```diff
@@ -1888,6 +1888,7 @@ The Druid SQL server is configured through the following properties on the Broke
 |`druid.sql.avatica.minRowsPerFrame`|Minimum acceptable value for the JDBC client `Statement.setFetchSize` method. The value for this property must greater than 0. If the JDBC client calls `Statement.setFetchSize` with a lesser value, Druid uses `minRowsPerFrame` instead. If `maxRowsPerFrame` is less than `minRowsPerFrame`, Druid uses the minimum value of the two. For handling queries which produce results with a large number of rows, you can increase this value to reduce the number of fetches required to completely transfer the result set.|100|
 |`druid.sql.avatica.maxStatementsPerConnection`|Maximum number of simultaneous open statements per Avatica client connection.|4|
 |`druid.sql.avatica.connectionIdleTimeout`|Avatica client connection idle timeout.|PT5M|
+|`druid.sql.avatica.fetchTimeoutMs`|Avatica fetch timeout, in milliseconds. When a request for the next batch of data takes longer than this time, Druid returns an empty result set, causing the client to poll again. This avoids HTTP timeouts for long-running queries. |5000|
```

Comment:
Can there be a suggestion for a good value that one (who is configuring this property) can set based on their querying patterns? For example, something like `Set it to something which is lesser than the estimated time between two consecutive fetches`.

Reply:
This is a tricky one! The value should be less than the minimum HTTP timeout anywhere in the request chain: client, proxy server, gateway/firewall, Router and Broker. It is hard to know these. The default is 5 sec, which should be well under any reasonable timeout, without impacting performance. (The timeout happens only when the query is slow, so we're outside the high-performance realm already.)

Added a comment that the default of 5 sec. should be fine in most cases.